pages/firebase-outputs.js

```diff
@@ -0,0 +1,82 @@
+import { useEffect, useState } from 'react';
+import Link from 'next/link';
+
+import {
+  getListOfAllArticlesByAlphabet,
+  getListOfAllArticlesByDate,
+  getListOfLastArticles,
+} from '../firebase/api/articles';
+
+const FirebaseFirst = () => {
+  const [articleList, setArticleList] = useState([]);
+  const [latestArticles, setLatestArticles] = useState([]);
+  const [alphabeticalListOfArticles, setAlphabeticalListOfArticles] = useState(
+    [],
+  );
+
+  useEffect(() => {
+    const listOfArticles = getListOfAllArticlesByDate();
+    listOfArticles.then((value) => setArticleList(value));
+
+    const listOfLastestArticles = getListOfLastArticles();
+    listOfLastestArticles.then((value) => setLatestArticles(value));
+
+    const listOfArticlesAlphabet = getListOfAllArticlesByAlphabet();
+    listOfArticlesAlphabet.then((value) =>
+      setAlphabeticalListOfArticles(value),
+    );
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <div className="homepage">
+      <div>
+        <h2 className="headline--1 mb-4">All articles</h2>
+        {articleList.map(({ id, link, title, dateOfPublication }) => (
+          <div className="m-5" key={id}>
+            <Link href={`/article/${link}`}>
+              <a className="text-blue-600 underline">
+                <h2>{title}</h2>
+              </a>
+            </Link>
+            <p>Publikováno dne: {dateOfPublication}</p>
+          </div>
+        ))}
+      </div>
+
+      <div>
+        <h2 className="headline--1 mb-4">Latest 3 articles</h2>
+        {latestArticles.map(({ id, link, title, dateOfPublication }) => (
+          <div className="m-5" key={id}>
+            <Link href={`/article/${link}`}>
+              <a className="text-blue-600 underline">
+                <h2>{title}</h2>
+              </a>
+            </Link>
+            <p>Publikováno dne: {dateOfPublication}</p>
+          </div>
+        ))}
+      </div>
+
+      <div className="m-5">
+        <h2 className="headline--1 mt-8">Alphabetical List</h2>
+        {Object.entries(alphabeticalListOfArticles).map((item) => (
+          <div key={item[0]}>
+            <h3 className="headline--4">{item[0]}</h3>
+            {item[1].map(({ title, link, id }) => (
+              <li className="my-2" key={id}>
+                <Link href={`/article/${link}`}>
+                  <a className="block text-blue-600 underline">
+                    <h2>{title}</h2>
+                  </a>
+                </Link>
+              </li>
+            ))}
+          </div>
+        ))}
```

Comment:
This seems confusing. Let's open it up on the call.

Reply:
`{Object.entries(alphabeticalListOfArticles).map(([groupLetter, articles]) => (`